### DISCUSSION TITLE
New idea/feature power saving deep sleep when idle

### DISCUSSION THREAD

**Discussion by virhao** [dl9rdz/rdz_ttgo_sonde#485] [TTGO firmware]
I have idea to add option for downloading actual time from internet and then esp32 could go to sleep in selected hours to save power especially when working only on battery . Example there only 2 hours a day  12:00 14:00  when radiosondes could be received in my area  and rest device is wasting power doing nothing it would be nice to select working hours between 12:00 and 14:00 and when it's not working could go to deep sleep with timer to wake up to next selected time

**Comment by ckuethe** [comment]
Time is already [set via NTP](https://github.com/dl9rdz/rdz_ttgo_sonde/blob/206ac3eee7e7ea4a5e3920bad03bb4e8a9c87ebb/RX_FSK/RX_FSK.ino#L2395) in the `enableNetwork()` function.

Aside from putting the ESP32 to sleep, you'll also want to power down the GPS and the LoRa module. There's also a `setPowerSave()` function in u8g2 graphics library. 

https://hackaday.io/project/171304-esp32-lora-gps-bike-thing/log/176718-power-up-and-gps has some info about controlling the AXP192.
